src/disc/Albany_DiscretizationFactory.cpp

```diff
@@ -105,12 +105,14 @@ DiscretizationFactory::createMeshStruct(Teuchos::RCP<Teuchos::ParameterList> dis
         int extruded_ws_size = disc_params->get("Workset Size", 50);
         int basal_ws_size = -1;
         if(extruded_ws_size != -1) {
-          basal_ws_size =  extruded_ws_size/ (disc_params->get<int>("NumLayers") * ((disc_params->get<std::string>("Element Shape") == "Tetrahedron") ? 3 : 1));
+          const int num_cells_per_layer = disc_params->get<std::string>("Element Shape") == "Tetrahedron" ? 3 : 1;
```

Comment:
The name is a bit misleading, since it sounds like it's the number of cells over a whole horizontal layer, rather than the number of 3d cells in one layer of an extruded single 2d cell. That said, I think finding a self-explanatory name is quite hard (unless one is fine with 100chars-long names). But maybe we could clarify it with a comment?

Reply:
Yeah, I was trying to make this more clear but I failed. Will add the comment.